recipes/ceres-solver/all/conanfile.py

```diff
@@ -0,0 +1,112 @@
+            self._cmake.definitions["GFLAGS"] = self.options.use_gflags
+            self._cmake.definitions["BUILD_EXAMPLES"] = False           #Requires gflags
+            self._cmake.definitions["BUILD_TESTING"] = False            #Requires gflags
+            self._cmake.definitions["BUILD_DOCUMENTATION"] = False      #Requires python modules Sphinx and sphinx-rtd-theme
+            self._cmake.definitions["CUSTOM_BLAS"] = self.options.use_custom_blas
+            self._cmake.definitions["EIGEN_PREFER_EXPORTED_EIGEN_CMAKE_CONFIGURATION"] = False    #Set to false to Force CMake to use the conan-generated dependencies
+            self._cmake.definitions["GLOG_PREFER_EXPORTED_GLOG_CMAKE_CONFIGURATION"] = False      #Set to false to Force CMake to use the conan-generated dependencies
+            self._cmake.definitions["GFLAGS_PREFER_EXPORTED_GFLAGS_CMAKE_CONFIGURATION"] = False  #Set to false to Force CMake to use the conan-generated dependencies
+            self._cmake.definitions["EIGENSPARSE"] = self.options.use_eigen_sparse
+            self._cmake.definitions["SUITESPARSE"] = False  #Optional. Not sufpported right now because SuiteSparse is not part of conan-index
+            self._cmake.definitions["LAPACK"] = False       #Optional. Not supported right now because LAPACK is not part of conan-index
+            self._cmake.definitions["OPENMP"] = False
+            self._cmake.definitions["CXSPARSE"] = False     #Optional. Not supported right now because CXSSPARSE is not part of conan-index
+            self._cmake.definitions["MINIGLOG"] = not self.options.use_glog
+            self._cmake.definitions["TBB"] = self.options.use_TBB
+            self._cmake.definitions["CXX11_THREADS"] = self.options.use_CXX11_threads
+            self._cmake.definitions["CXX11"] = self.options.use_CXX11
+            self._cmake.definitions["SCHUR_SPECIALIZATIONS"] = self.options.use_schur_specializations
+            if self.settings.os == "Windows":
+                self._cmake.definitions["MSVC_USE_STATIC_CRT"] = str(self.settings.compiler.runtime) == "MT" or str(self.settings.compiler.runtime) == "MTd"
+            self._cmake.configure()
+        return self._cmake
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.remove("fPIC")
+        if self.settings.build_type == "Debug" and self.options.use_glog:
+            raise ConanInvalidConfiguration("Ceres-solver only links against the release version of glog")
+        if self.options.use_glog and not self.options.use_gflags: #At this stage we can't check the value of self.options["glog"].with_gflags so we asume it is true because is the default value
+            raise ConanInvalidConfiguration("To depend on glog built with gflags (Default behavior) set use_gflags=True, otherwise Ceres may fail to link due to missing gflags symbols. ")
```

Comment:
```suggestion
            raise ConanInvalidConfiguration("To depend on glog built with gflags (Default behavior) set use_gflags=True, otherwise Ceres may fail to link due to missing gflags symbols. ")

    def configure(self):
        if self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version) < "16":
            raise ConanInvalidConfiguration("Ceres Solver requires Visual Studio >=16")
```

Reply:
Why do you want to ditch VS14 too? The bug is present only on VS15 without the latest updates.

Reply:
I see, let's drop VC15 only so